src/getstream/updateBetterChannelMembers.js

```diff
@@ -0,0 +1,176 @@
+      memberDataFromDb;
+
+    const updatedUsername = is_anonymous
+      ? `${anon_user_info_color_name} ${anon_user_info_emoji_name}`
+      : username;
+
+    newChannelName += `${updatedUsername}${SEPARATOR}`;
+
+    const defaultUser = {
+      ...member,
+      is_anonymous,
+      user: {
+        ...member.user,
+        is_anonymous: is_anonymous,
+        username: updatedUsername,
+        name: updatedUsername
+      }
+    };
+
+    if (is_anonymous) {
+      defaultUser.is_anonymous = is_anonymous;
+      const anon_user_info = __getAnonUserInfoPriority(memberDataFromDb, member);
+      defaultUser.anon_user_info_color_name = anon_user_info.color_name;
+      defaultUser.anon_user_info_color_code = anon_user_info.color_code;
+      defaultUser.anon_user_info_emoji_name = anon_user_info.emoji_name;
+      defaultUser.anon_user_info_emoji_code = anon_user_info.emoji_code;
+    }
+
+    return defaultUser;
+  });
+
+  return {
+    better_channel_member,
+    new_channel_name: newChannelName.slice(0, -SEPARATOR.length)
+  };
+};
+
+/**
+ * @typedef {Object} UpdateBetterChannelMember
+ * @property {import('stream-chat').ChannelMember[]} betterChannelMember
+ * @property {string} newChannelName
+ * @property {Object} betterChannelMemberObject
+ * @property {import("stream-chat").ChannelAPIResponse} updatedChannel
+ */
+
+/**
+ *
+ * @param {import('stream-chat').Channel} chatChannel
+ * @param {import("stream-chat").ChannelAPIResponse} channel
+ * @param {Partial<import("stream-chat").ChannelResponse>} additionalUpdateData
+ * @returns {Promise<UpdateBetterChannelMember>}
+ */
+const updateBetterChannelMembers = async (
+  chatChannel,
+  channel,
+  withUpdate = false,
+  additionalUpdateData = {}
+) => {
+  if (!chatChannel) throw new Error('Chat Channel is required');
+  if (!channel) throw new Error('Channel is required');
+
+  const {members} = channel;
+
+  const membersIds = members.map((member) => member.user_id);
+
+  const membersDataFromDb = await getAllChatAnonimityUserInfo(channel?.channel?.id, membersIds);
+
+  const membersDataFromDbMap = _.keyBy(membersDataFromDb, 'user_id');
+
+  const {better_channel_member, new_channel_name} = __helperProcessBetterChannelMember(
+    members,
+    membersDataFromDbMap
+  );
+
+  const checkedChannelName = channel?.channel?.is_name_custom
+    ? channel?.channel?.name
+    : new_channel_name;
+
+  const defaultUpdateData = {
+    better_channel_member,
+    name: checkedChannelName,
+    ...additionalUpdateData
+  };
+
+  if (withUpdate) {
+    try {
+      channel = await chatChannel.updatePartial({
+        set: defaultUpdateData
+      });
+    } catch (e) {
+      console.error('Failed to update channel', e);
+    }
+  }
+
+  return {
+    betterChannelMember: better_channel_member,
+    newChannelName: checkedChannelName,
+    betterChannelMemberObject: _.keyBy(better_channel_member, 'user_id'),
+    updatedChannel: channel
+  };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `updateBetterChannelMembers` function, there's no error handling for when the `chatChannel.updatePartial` call fails. If an error occurs during this operation, it will be logged but the function will still return as if nothing happened. This could lead to unexpected behavior. Consider adding proper error handling here.

```diff
-    try {
-      channel = await chatChannel.updatePartial({
-        set: defaultUpdateData
-      });
-    } catch (e) {
-      console.error('Failed to update channel', e);
-    }
+    try {
+      channel = await chatChannel.updatePartial({
+        set: defaultUpdateData
+      });
+    } catch (e) {
+      console.error('Failed to update channel', e);
+      throw new UpdateChannelError('Failed to update channel', e);
+    }
```
Again, `UpdateChannelError` would be a custom error class that you'd need to define.


<!-- This is an auto-generated comment by OSS CodeRabbit -->